bootstrap.go

```
@@ -42,15 +43,32 @@ func Start(ctx context.Context, opts ...Option) (StopFunc, error) {
 
 	l, err := datadogLogger.NewLogger(datadogLogger.WithGlobalLogger())
 	if err != nil {
-		return noop, fmt.Errorf("Failed to initialize the Datadog logger: %w", err)
+		return noop, fmt.Errorf("failed to initialize the Datadog logger: %w", err)
 	}
 	ddtrace.UseLogger(l)
 
+	// Make sure stop() only runs once
+	var once sync.Once
+	var cancelErr error
 	cancel := func() error {
-		return stop(options)
+		once.Do(func() {
+			cancelErr = stop(options)
+		})
+		return cancelErr
 	}
 
 	err = start(options)
+
+	if err != nil {
+		return noop, err
+	}
+
+	// Clean up if the context is cancelled
+	go func() {
+		<-ctx.Done()
+		cancelErr = cancel()
```

This assignment is superfluous because the value will already have been set in the cancel function itself. That bugs me.

If I remove it, we run into a validation error for not checking the err value.

---

You can add a comment that this context-listening is just used to early-start the flushing when the context is cancelled, and then discard the error (`_ = cancel()`) and add a nolint-comment to pass validation

---

You ~can~ must discard the error, to avoid a race-condition where this shared error-variable is being written (with the same value) at the same time as it is being read from another goroutine.